### PR TITLE
V1-202 - notifications

### PR DESCRIPTION
**Pull request by zhuk-vezunchikz**
- The manager shouldn't get the notification when passed the test
- Notifications sorted from new to old (top to bottom)
- Added the name of the course and name of the employee in the notifications
- Only last 10 notification items displayed
- Notifications are red until the window closes

### CODE REVIEW

File: frontend/src/components/NotificationsBar/Notification/Notification.tsx [hunk 1]
```diff
@@ -12,16 +12,24 @@ interface NotificationProps {
 
 const NotificationPlate: React.FC<NotificationProps> = ({ note }) => {
```
Comment (irina-sinitsyna) [inline frontend/src/components/NotificationsBar/Notification/Notification.tsx:13]: file and component inside should have the same name. so please rename file, folder and component to Notification or NotificationPlate

Comment (irina-sinitsyna) [reply frontend/src/components/NotificationsBar/Notification/Notification.tsx:13]: and also here you should add an index.ts file with default export in Notification folder and NotificationsBar folder

Comment (irina-sinitsyna) [inline frontend/src/components/NotificationsBar/Notification/Notification.tsx:13]:
```suggestion
const NotificationPlate: FC<NotificationProps> = ({ note }) => {
```

File: frontend/src/components/NotificationsBar/Notification/Notification.tsx [hunk 2]
```diff
@@ -12,16 +12,24 @@ interface NotificationProps {
...
   const isOldNotification = note.status === NOTIFICATION_STATUSES.old;
+  const isUserNotifications = note.type === NOTIFICATION_TYPES.user;
```
Comment (irina-sinitsyna) [inline frontend/src/components/NotificationsBar/Notification/Notification.tsx:16]: const {courseName, description, title, userName} = note || {}

File: frontend/src/components/NotificationsBar/Notification/Notification.tsx [hunk 3]
```diff
@@ -12,16 +12,24 @@ interface NotificationProps {
...
+        <div>
+          <Title>{`${note.courseName} is ${note.title}`}</Title>
+          <Description>{note.description}</Description>
+        </div>
+      ) : (
+        <div>
+          <Title>{note.userName}</Title>
+          <Description>{`${note.description} ${note.courseName}`}</Description>
+        </div>
+      )}
```
Comment (irina-sinitsyna) [inline frontend/src/components/NotificationsBar/Notification/Notification.tsx:32]:
```suggestion
      {isUserNotifications ? (
        <div>
          <Title>{`${courseName} is ${title}`}</Title>
          <Description>{description}</Description>
        </div>
      ) : (
        <div>
          <Title>{userName}</Title>
          <Description>{`${description} ${courseName}`}</Description>
        </div>
      )}
```

Comment (irina-sinitsyna) [reply frontend/src/components/NotificationsBar/Notification/Notification.tsx:32]: and also maybe <div> can be replaced with empty tag in this case?

File: frontend/src/components/NotificationsBar/Notification/Notification.tsx [hunk 4]
```diff
@@ -12,16 +12,24 @@ interface NotificationProps {
...
   return (
     <NotificationContainer isOld={isOldNotification}>
       <ImageWrapper>
         <img alt="notification" src={isOldNotification ? warningIcon : redWarningIcon} />
       </ImageWrapper>
-      <div>
-        <Title>{note.title}</Title>
-        <Description>{note.description}</Description>
-      </div>
+      {isUserNotifications ? (
```
Comment (ostapdans99) [inline frontend/src/components/NotificationsBar/Notification/Notification.tsx:22]: for what you are changed layout? You can change only content inside Title and Description

**Review by ostapdans99 — APPROVED** [review]

File: frontend/src/components/NotificationsBar/Notification/Notification.tsx (hunk 1 above)
Comment (zhuk-vezunchikz) [reply frontend/src/components/NotificationsBar/Notification/Notification.tsx:13]: We already have index files in these folders

File: frontend/src/components/NotificationsBar/Notification/Notification.tsx (hunk 3 above)
Comment (zhuk-vezunchikz) [reply frontend/src/components/NotificationsBar/Notification/Notification.tsx:32]: No, our notification style is breaking down

File: frontend/src/components/NotificationsBar/Notification/Notification.tsx [hunk 5]
```diff
@@ -1,29 +1,37 @@
...
 import { warningIcon, redWarningIcon } from 'icons';
-import { Notification } from 'types/notification';
+import { INotification } from 'types/INotification';
 
 import { NotificationContainer, ImageWrapper, Title, Description } from './styled';
 
 interface NotificationProps {
-  note: Notification;
+  note: INotification;
 }
```
Comment (irina-sinitsyna) [inline frontend/src/components/NotificationsBar/Notification/Notification.tsx:11]: please move all the interfaces to separate file types.ts inside NotificationsBar folder

Comment (irina-sinitsyna) [reply frontend/src/components/NotificationsBar/Notification/Notification.tsx:11]: and also they should be renamed from Prips/IProps to INotificationBarContainerProps and so on

File: backend/src/interfaces/Ientities/Itest.ts [hunk 6]
```diff
@@ -21,7 +21,7 @@ interface ITestAnswer {
   aN: number;
 }
 
-type TAchievments = {
+type TAAchievements = {
```
Comment (dapodev) [inline backend/src/interfaces/Ientities/Itest.ts:24]: typo?

File: backend/src/utils/fullNameCreator.ts [hunk 7]
```diff
@@ -0,0 +1 @@
+export const fullNameCreator = (firstName: string, lastName: string) => `${firstName} ${lastName}`;
```
Comment (dapodev) [inline backend/src/utils/fullNameCreator.ts:1]: pls, make name of the function to sound like a verb. Usually classes are named as nouns.

Comment (dapodev) [reply backend/src/utils/fullNameCreator.ts:1]: `combineFullName` for example

**Review by dapodev — CHANGES_REQUESTED** [review]

File: frontend/src/components/NotificationsBar/NotificationPlate/NotificationPlate.tsx [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+
+import { NOTIFICATION_STATUSES, NOTIFICATION_TYPES } from 'constants/statuses';
+import { INotification } from 'types/Notification';
+import { warningIcon, redWarningIcon } from 'icons';
+
+import { NotificationContainer, ImageWrapper, Title, Description } from './styled';
+
+interface NotificationProps {
+  note: INotification;
+}
```
Comment (irina-sinitsyna) [inline frontend/src/components/NotificationsBar/NotificationPlate/NotificationPlate.tsx:11]: please move to types.ts

File: frontend/src/components/NotificationsBar/NotificationsBar.tsx [hunk 9]
```diff
@@ -5,8 +5,9 @@ import { NO_NOTIFICATIONS } from 'constants/messages';
 import { alertIcon } from 'icons';
 
 import { INotificationsProps } from './types';
-import NotificationPlate from './Notification';
+
```
Comment (irina-sinitsyna) [inline frontend/src/components/NotificationsBar/NotificationsBar.tsx:8]: extra empty line

File: frontend/src/components/NotificationsBar/styled.ts [hunk 10]
```diff
@@ -55,10 +55,7 @@ export const RedMark = styled('div')({
...
   backgroundColor: '#D43E41',
-  position: 'relative',
-  top: '-18px',
-  right: '30px',
-  [theme.breakpoints.down('sm')]: {
-    top: '-15px',
-  },
+  position: 'absolute',
+  top: '0px',
+  right: '0px',
```
Comment (irina-sinitsyna) [inline frontend/src/components/NotificationsBar/styled.ts:60]:
```suggestion
  top: 0,
  right: 0,
```

**Review by dapodev — APPROVED** [review]

File: frontend/src/components/NotificationsBar/NotificationPlate/NotificationPlate.tsx [hunk 11]
```diff
@@ -1,14 +1,10 @@
...
 import { NOTIFICATION_STATUSES, NOTIFICATION_TYPES } from 'constants/statuses';
-import { INotification } from 'types/Notification';
 import { warningIcon, redWarningIcon } from 'icons';
 
 import { NotificationContainer, ImageWrapper, Title, Description } from './styled';
-
-interface NotificationProps {
-  note: INotification;
-}
+import { NotificationProps } from '../types';
```
Comment (irina-sinitsyna) [inline frontend/src/components/NotificationsBar/NotificationPlate/NotificationPlate.tsx:7]: import { INotificationPlateProps } from '../types';

**Review by dapodev — APPROVED** [review]

**Review by irina-sinitsyna — APPROVED** [review]

**Review by dapodev — APPROVED** [review]

**Review by dapodev — APPROVED** [review]

**Pull request merged**